Lib/glyphsLib/classes.py

```diff
@@ -2732,3 +2723,13 @@ def note(self):
     @note.setter
     def note(self, value):
         self.customParameters["note"] = value
+
+    # FIXME: (jany) I don't understand why this is needed for
+    #   gridSubDivisions but it works without it for upm
+    @property
+    def gridSubDivisions(self):
+        return self.gridSubDivision
```

Comment:
A property getter cannot recursively return itself, nor should the setter overwrite itself..
Either you get or set something else (e.g. `self._gridSubDivision`) or simply remove the property if it's a simple read/write attribute.

Reply:
woah.. I totally missed that "s" at the end. Please ignore my comment, sorry :)

Reply:
No problem :) Do you know why there is a difference between `upm` vs `unitsPerEm` (which don't have the getter/setter) and this `gridSubDivision(s)` which need a getter/setter for the tests to pass?

Reply:
i think it's because in the GSBase constructor, we are not translating the key from `_classesForName` dict using the `_wrapperKeysTranslate` dict, so `gridSubDivision` (without the "s") is being set.

Reply:
I suggest you add a line `key = self._wrapperKeysTranslate.get(key, key)` before calling `setattr(self, key, value)` in the `GSBase.__init__`, like the `__setitem__` method does..

Although I really don't like that one says to a dict, "please set this key-value pair", and the dict replies, "Ok, i'll set the value but using a slightly different key".
Also, I've never seen a write-only dictionary, with a `__setitem__` but no corresponding `__getitem__`...

Reply:
Btw, the `__contains__` method also does not do the _wrapperKeysTranslate trick, but simply delegates to `hasattr`.
So one may set `font["unitsPerEm"] = 1000`, and it will set the `upm` attribute to 1000, but then if you test `"unitsPerEm" in font` it will return `False`. This is very weird.

Reply:
The GlyphsSDK seems to have `font.unitsPerEm()` which returns the objc font value for unitsPerEm, while `font.upm` is the Python API wrapper around that which can be set.

It’s similar for`gridSubDivision`, it’s the obj font function whereas `gridSubDivisions` is the Python API wrapper around `gridSubDivision()` and `setGridSubDivision(value)`.

Reply:
Maybe we should stick to only having `font.upm` and `font.gridSubDivisions` and not have `font.unitsPerEm` nor `font.gridSubDivision` in the Python API.

/cc @schriftgestalt

Reply:
yes, it doesn't make sense to have both, at least here in glyphsLib.